apps/api/R/inputs.R

```diff
@@ -0,0 +1,116 @@
+library(dplyr)
+
+#' Search for Inputs containing wildcards for filtering
+#' @param model_id Model Id (character)
+#' @param site_id Site Id (character)
+#' @param offset
+#' @param limit 
+#' @return Information about Inputs based on model & site
+#' @author Tezan Sahu
+#* @get /
+searchInputs <- function(req, model_id=NULL, site_id=NULL, offset=0, limit=50, res){
+  if (! limit %in% c(10, 20, 50, 100, 500)) {
+    res$status <- 400
+    return(list(error = "Invalid value for parameter"))
+  }
+  
+  dbcon <- PEcAn.DB::betyConnect()
+  
+  inputs <- tbl(dbcon, "inputs") %>%
+    select(input_name=name, id, site_id, format_id, start_date, end_date)
+  
+  inputs <- tbl(dbcon, "dbfiles") %>%
+    select(file_name, file_path, container_type, id=container_id, machine_id) %>%
+    inner_join(inputs, by = "id") %>%
+    filter(container_type == 'Input') %>%
+    select(-container_type)
+  
+  inputs <- tbl(dbcon, "machines") %>%
+    select(hostname, machine_id=id) %>%
+    inner_join(inputs, by='machine_id') %>%
+    select(-machine_id)
+  
+  inputs <- tbl(dbcon, "modeltypes_formats") %>%
+    select(tag, modeltype_id, format_id, input) %>%
+    inner_join(inputs, by='format_id') %>%
+    filter(input) %>%
+    select(-input)
+  
+  inputs <- tbl(dbcon, "formats") %>%
+    select(format_id = id) %>%
```

Comment:
I think it would be useful to also grab the format name and mimetype, as that seems really relevant to someone looking  to pull data.

Reply:
Cool...I'll look into it